classes/Product.php

```diff
@@ -4764,12 +4764,8 @@ public static function cacheFrontFeatures($product_ids, $id_lang)
      *
      * @return array|false Matching products
      */
-    public static function searchByName($id_lang, $query, Context $context = null, $limit = null)
+    public static function searchByName($id_lang, $query, $limit = null)
```

Comment:
Hi @idnovate I agree with you that this variable, if unused, is completely useless ; however by removing it we introduce a risk.

Let's say we merge this PR in `develop` : that will be delivered in PrestaShop 1.7.8

If today a module uses `Product::searchByName($id_lang, $query, null, $limit);` with PS 1.7.7 it works well.
Tomorrow the shop upgrades from 1.7.7 to 1.7.8 and now `Product::searchByName($id_lang, $query, null, $limit);` has a very different meaning as `$limit` argument is now the 3rd arg, not the 4th.

So I think we can _deprecate_ this parameter by adding a phpDoc statement, or even a warning. But I dont think the value of removing this useless argument is higher than the cost of breaking modules relying on this function.

I think the BC break should be avoided here.

If however `develop` was meant to be released as a new major version, where BC breaks are allowed, that would be totally legit 😄  .

Reply:
I agree with you. So I have just removed the `if` inside the function, but keep the arguments.